packages/flutter_tools/lib/src/build_system/targets/linux.dart

```diff
@@ -86,62 +83,86 @@ class UnpackLinux extends Target {
   }
 }
 
-/// Creates a debug bundle for the Linux desktop target.
-class DebugBundleLinuxAssets extends Target {
+class DebugBundleLinuxAssets extends ApplicationAssetBundle {
   const DebugBundleLinuxAssets();
 
+  @override
+  List<Source> get inputs => <Source>[
+    ...super.inputs,
+    const Source.artifact(Artifact.vmSnapshotData, mode: BuildMode.debug),
+    const Source.artifact(Artifact.isolateSnapshotData, mode: BuildMode.debug),
```

Comment:
Why are the inputs and outputs in the subclass, but the logic that copies from the input to the output location in the superclass? Shouldn't the superclass be describing the inputs and outputs it's managing?

Reply:
The `vm_snapshot_data` , `isolate_snapshot_data`, and `kernel_blob.bin` are not part of profile/release builds. This sort of structure allows me to statically describe all of the dependencies while only having a single `build` implementation.

Reply:
I guess I'm not seeing how having one `build` method is an advantage when it means having the input/output description in a different class—and file—than the code that actually causes it to be an input/output. It seems like this would be incredibly error-prone to maintain because it's not at all obvious that a change here would require changes in a different file, or the other way around.

Why not have a debug subclass that has the debug inputs, outputs, and build logic, and then derive from that here?